sdk/trace/batch_span_processor.go

```diff
@@ -378,6 +388,9 @@ func (bsp *batchSpanProcessor) enqueueBlockOnQueueFull(ctx context.Context, sd R
 		return true
 	default:
 		atomic.AddUint32(&bsp.dropped, 1)
+		if bsp.o.DroppedSpansCallback != nil {
+			go bsp.o.DroppedSpansCallback()
```

Comment:
I would prefer this to be a blocking call, one the user is responsible for making very quickly. I also want to see OTel specify a conventional metric to count these drops, but don't want to block you here.